apps/journeys-admin/src/components/Editor/Toolbar/Toolbar.tsx

```diff
@@ -120,27 +139,78 @@ export function Toolbar(): ReactElement {
             </Button>
           </Tooltip>
 
-          <Stack flexGrow={1} sx={{ minWidth: 0 }}>
-            <Typography
-              variant="subtitle1"
-              sx={{
-                overflow: 'hidden',
-                whiteSpace: 'nowrap',
-                textOverflow: 'ellipsis'
-              }}
-            >
-              {journey.title}
-            </Typography>
-            <Typography
-              variant="caption"
+          <Stack flexGrow={1} flexShrink={1} sx={{ minWidth: 0 }}>
+            <Box
+              flexShrink={1}
               sx={{
+                display: 'inline-flex',
                 overflow: 'hidden',
                 whiteSpace: 'nowrap',
                 textOverflow: 'ellipsis'
               }}
             >
-              {journey.description}
-            </Typography>
+              <Tooltip
+                title="Click to edit"
+                placement="bottom"
+                arrow
+                PopperProps={{
+                  modifiers: [
+                    {
+                      name: 'offset',
+                      options: {
+                        offset: journey.description === '' ? [0, 2] : [0, -10.3]
+                      }
+                    }
+                  ]
+                }}
+              >
+                <Button
+                  variant="text"
+                  onClick={handleSelectTitle}
+                  color="secondary"
+                  sx={{
+                    maxWidth: 'auto',
+                    display: 'block',
+                    textAlign: 'left',
+                    overflow: 'hidden',
+                    whiteSpace: 'nowrap',
+                    textOverflow: 'ellipsis',
+                    borderRadius: '8px',
+                    flexShrink: 1
+                  }}
+                >
+                  <Typography
+                    variant="subtitle1"
+                    sx={{
+                      overflow: 'hidden',
+                      whiteSpace: 'nowrap',
+                      textOverflow: 'ellipsis',
+                      flexShrink: 1
+                    }}
+                  >
+                    {journey.title}
+                  </Typography>
+                  <Typography
+                    variant="caption"
+                    sx={{
+                      maxWidth: 'auto',
+                      overflow: 'hidden',
+                      whiteSpace: 'nowrap',
+                      textOverflow: 'ellipsis',
+                      flexShrink: 1
+                    }}
+                  >
+                    {journey.description}
+                  </Typography>
+                </Button>
+              </Tooltip>
+            </Box>
+            {journey?.id != null && (
+              <TitleDescriptionDialog
+                open={titleDialogOpen}
+                onClose={handleCloseTitle}
+              />
+            )}
```

Comment:
I think this check is redundant, `TitleDescriptionDialog` does its own journey null check.
```suggestion
              <TitleDescriptionDialog
                open={titleDialogOpen}
                onClose={handleCloseTitle}
              />
```